Automattic/wp-calypso
# Integrate the refer-a-friend eligibility check

## Proposed Changes

This PR is the calypso-side of the eligibility check by @gmovr in code-D154150. With the patch, the endpoint will respond with an error when the user isn't eligible to the refer-a-friend program. Since the current flow requires a POST request to enroll in the peer referral program, this PR handles the case by communicating the error through an error notice and disabling the button.

https://github.com/Automattic/wp-calypso/assets/1842898/e0d4a25a-00ac-41f4-89d6-2c4bf70d4401

A side clean-up is to remove the `/earn/refer-a-friend` component that's no longer in use for posterity.

## Why are these changes being made?


Mainly for a better UX accompanying code-D154150. 

## Testing Instructions



* Logged-in as a user not owning any paid plan
    * Go to /earn and click "Earn free credits". 
    * The CTA should be disabled upon the error notice showing up.
* Logged-in as a user owning a paid plan
    * Go to /earn and click "Earn free credits". 
    * The peer-referral link should be generated and show up as expected.

## Pre-merge Checklist
